Frameworks/Foundation/NSOperation.mm

```diff
@@ -131,139 +162,122 @@ - (void)setThreadPriority:(double)priority {
  @Status Interoperable
 */
 - (void)setCompletionBlock:(void (^)(void))block {
-    id oldBlock = priv->completionBlock;
-    priv->completionBlock = [block copy];
-    [oldBlock release];
+    std::lock_guard<std::recursive_mutex> lock(_completionBlockLock);
+    [self willChangeValueForKey:@"completionBlock"];
+    _completionBlock = block;
+    [self didChangeValueForKey:@"completionBlock"];
 }
 
 /**
  @Status Interoperable
 */
 - (void (^)(void))completionBlock {
-    return priv->completionBlock;
+    std::lock_guard<std::recursive_mutex> lock(_completionBlockLock);
+    return [[_completionBlock retain] autorelease];
 }
 
 /**
  @Status Interoperable
 */
-- (BOOL)isReady {
-    //  Note, check dependencies when we get them
-    int count = [priv->dependencies count];
-
-    for (int i = 0; i < count; i++) {
-        id op = [priv->dependencies objectAtIndex:i];
-
-        if (![op isFinished]) {
-            return NO;
-        }
-    }
-    return YES;
+- (BOOL)isCancelled {
+    return _cancelled;
 }
 
 /**
  @Status Interoperable
 */
-- (BOOL)isCancelled {
-    return priv->cancelled != 0;
+- (BOOL)isFinished {
+    return _finished;
 }
 
 /**
  @Status Interoperable
 */
-- (BOOL)isFinished {
-    return priv->finished != 0;
+- (BOOL)isExecuting {
+    return _executing;
 }
 
 /**
  @Status Interoperable
 */
-- (BOOL)isExecuting {
-    return priv->executing != 0;
+- (BOOL)isConcurrent {
+    return NO;
```

Comment:
It's the _subclass_ that's concurrent; NSOperation has to have the machinery to support both even though it is itself nonconcurrent.